### PR TITLE
[Testing:Autograding] Add first testcases for submitty autograding shipper

### PR DESCRIPTION
**Pull request by emaicus**
Adds the first unit tests to the submitty autograding shipper. These test the shipper's ability to "short circuit" testcase execution. Setup and teardown functions are included for the test suite to facilitate future tests.

### CODE REVIEW

File: autograder/tests/test_submitty_autograding_shipper.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+import json
+import shutil
+import sys
+import unittest
+import contextlib
+import copy
+
+# Done to avoid making the base autograder directory a module.
+# This import must be done inside of the class for the mock to take affect
+sys.path.append("..")
```
Comment (hdrodz97) [inline autograder/tests/test_submitty_autograding_shipper.py:11]:
```suggestion
```

This isn't necessary as long as we run the tests from within the `autograding/` directory which, judging from [.travis.yml](https://github.com/Submitty/Submitty/blob/master/.travis.yml#L155-L157), we are doing. Running `coverage -m unittest discover` from `autograding/` runs fine on my VM without this line.

File: autograder/tests/test_submitty_autograding_shipper.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+# Holds all system config files (equivalent to /usr/local/submitty/config)
+CONFIG_DIR = os.path.join(TEST_ENVIRONMENT, 'config')
+SUBMITTY_DATA_DIR = os.path.join(TEST_ENVIRONMENT, 'autograding')
+
+# Autograding directories
+TODO_DIR = os.path.join(SUBMITTY_DATA_DIR, 'autograding_TODO')
+DONE_DIR = os.path.join(SUBMITTY_DATA_DIR, 'autograding_DONE')
+TO_BE_GRADED = os.path.join(SUBMITTY_DATA_DIR, 'to_be_graded_queue')
+GRADING = os.path.join(SUBMITTY_DATA_DIR, "grading")
```
Comment (hdrodz97) [inline autograder/tests/test_submitty_autograding_shipper.py:30]: Nit - this was renamed to `in_progress_grading`

File: autograder/tests/test_submitty_autograding_shipper.py (hunk 1 above)
Comment (emaicus) [reply autograder/tests/test_submitty_autograding_shipper.py]: Removed

File: autograder/tests/test_submitty_autograding_shipper.py (hunk 2 above)
Comment (emaicus) [reply autograder/tests/test_submitty_autograding_shipper.py]: Good catch, updated.

**Pull request merged**